e2e/jail/jail_test.go

```diff
@@ -27,60 +32,74 @@ func TestJailTestSuite(t *testing.T) {
 }
 
 type JailTestSuite struct {
-	suite.Suite
-	jail common.JailManager
+	e2e.NodeManagerTestSuite
+	Jail common.JailManager
 }
 
 func (s *JailTestSuite) SetupTest() {
-	s.jail = jail.New(nil)
-	s.NotNil(s.jail)
+	s.NodeManager = node.NewNodeManager()
+	s.Jail = jail.New(s.NodeManager)
 }
 
-func (s *JailTestSuite) TestInit() {
+func (s *JailTestSuite) TearDownTest() {
+	s.Jail.Stop()
+}
+
+func (s *JailTestSuite) TestInitWithoutBaseJS() {
 	errorWrapper := func(err error) string {
 		return `{"error":"` + err.Error() + `"}`
 	}
 
 	// get cell VM w/o defining cell first
-	cell, err := s.jail.Cell(testChatID)
+	cell, err := s.Jail.Cell(testChatID)
 
-	s.EqualError(err, "cell["+testChatID+"] doesn't exist")
+	s.EqualError(err, "cell '"+testChatID+"' not found")
 	s.Nil(cell)
 
 	// create VM (w/o properly initializing base JS script)
 	err = errors.New("ReferenceError: '_status_catalog' is not defined")
-	s.Equal(errorWrapper(err), s.jail.Parse(testChatID, ``))
+	s.Equal(errorWrapper(err), s.Jail.CreateAndInitCell(testChatID, ``))
 	err = errors.New("ReferenceError: 'call' is not defined")
-	s.Equal(errorWrapper(err), s.jail.Call(testChatID, `["commands", "testCommand"]`, `{"val": 12}`))
+	s.Equal(errorWrapper(err), s.Jail.Call(testChatID, `["commands", "testCommand"]`, `{"val": 12}`))
 
 	// get existing cell (even though we got errors, cell was still created)
-	cell, err = s.jail.Cell(testChatID)
+	cell, err = s.Jail.Cell(testChatID)
 	s.NoError(err)
 	s.NotNil(cell)
+}
 
+func (s *JailTestSuite) TestInitWithBaseJS() {
 	statusJS := baseStatusJSCode + `;
 	_status_catalog.commands["testCommand"] = function (params) {
 		return params.val * params.val;
 	};`
-	s.jail.BaseJS(statusJS)
+	s.Jail.SetBaseJS(statusJS)
 
 	// now no error should occur
-	response := s.jail.Parse(testChatID, ``)
+	response := s.Jail.CreateAndInitCell(testChatID, ``)
 	expectedResponse := `{"result": {"commands":{},"responses":{}}}`
 	s.Equal(expectedResponse, response)
 
 	// make sure that Call succeeds even w/o running node
-	response = s.jail.Call(testChatID, `["commands", "testCommand"]`, `{"val": 12}`)
+	response = s.Jail.Call(testChatID, `["commands", "testCommand"]`, `{"val": 12}`)
 	expectedResponse = `{"result": 144}`
 	s.Equal(expectedResponse, response)
 }
 
-func (s *JailTestSuite) TestParse() {
+func (s *JailTestSuite) TestMultipleInitError() {
+	response := s.Jail.CreateAndInitCell(testChatID, ``)
+	s.Equal(`{"error":"ReferenceError: '_status_catalog' is not defined"}`, response)
```

Comment:
This is badly designed. `CreateAndInitCell` is supposed to return JSON blob of `_status_catalog` variable. But this variable comes from the JS code passed to `CreateAndInitCell`. It should be a separate binding.

Reply:
How did it work earlier? :)

Reply:
It has never worked but in e2e tests, we were calling `Parse` with some dummy JS code. We could do it here as well and it might make more sense actually...

Reply:
I think let's leave this test with a comment that we need to fix this and make another test where it's called almost as before?

Reply:
The ultimate fix would be to get rid of this completely. status-react should verify `_status_catalog` variable as it comes from the base JS code or code they use to initialize a cell. It's not something that we provide. I will comment it in this way and create an issue to tackle it, ok?

Reply:
Yeah. I just want to cover the "normal" use case how it's used and the use case which shows an "incorrect" behavior which should just be visible to us all the time.